source/slang/hlsl.meta.slang

```diff
@@ -1367,13 +1364,84 @@ void TraceRay<payload_t>(
     RayDesc                         Ray,
     inout payload_t                 Payload);
 
+__target_intrinsic(glsl, "traceNVX")
+void __traceNVX(
+    RaytracingAccelerationStructure AccelerationStructure,
+    uint                            RayFlags,
+    uint                            InstanceInclusionMask,
+    uint                            RayContributionToHitGroupIndex,
+    uint                            MultiplierForGeometryContributionToHitGroupIndex,
+    uint                            MissShaderIndex,
+    float3                          Origin,
+    float                           TMin,
+    float3                          Direction,
+    float                           TMax,
+    int                             PayloadLocation);
+
+// TODO: Slang's parsing logic currently puts modifiers on
+// the `GenericDecl` rather than the inner decl when
+// using our default syntax, which seems wrong. We need
+// to fix this, but for now using the expanded `__generic`
+// syntax works in a pinch.
+//
+__generic<Payload>
+__target_intrinsic(glsl, "$XP")
+int __rayPayloadLocation(Payload payload);
+
+__generic<payload_t>
+__specialized_for_target(glsl)
+void TraceRay(
+    RaytracingAccelerationStructure AccelerationStructure,
+    uint                            RayFlags,
+    uint                            InstanceInclusionMask,
+    uint                            RayContributionToHitGroupIndex,
+    uint                            MultiplierForGeometryContributionToHitGroupIndex,
+    uint                            MissShaderIndex,
+    RayDesc                         Ray,
+    inout payload_t                 Payload)
+{
+    [__vulkanRayPayload]
+    static payload_t p;
+
+    p = Payload;
+    __traceNVX(
+        AccelerationStructure,
+        RayFlags,
+        InstanceInclusionMask,
+        RayContributionToHitGroupIndex,
+        MultiplierForGeometryContributionToHitGroupIndex,
+        MissShaderIndex,
+        Ray.Origin,
+        Ray.TMin,
+        Ray.Direction,
+        Ray.TMax,
+        __rayPayloadLocation(p));
+    Payload = p;
+}
+
 // 10.3.3
-bool ReportHit<attr_t>(float THit, uint HitKind, attr_t Attributes);
+bool ReportHit<A>(float tHit, uint hitKind, A attributes);

```

Comment:
The declarations here were originally trying to mimic exactly how the functions were declared in the provision DXR spec (version 0.91), which used `UperCamelCase` for value parameters, and `name_t` for type parameters.

I don't like that convention at all, so as I started to have to work with these functions more I started to massage the names to be more idomatic, but I also didn't push it through all the way because I don't like making larger diffs than strictly necessary.

One reason why it is worth talking about this is because I'd eventually like Slang to support using `key: value` syntax at call sites for by-name parameter matching (similar to how C# supports this feature). In that case the parameter names used in the standard library declarations would start to be part of the compatibility contract for the Slang compiler (parameter name changes would be source-breaking, binary-compatible changes).

So we eventually should decide whether the Slang standard library definitions of HLSL functionality should follow the MSDN/Microsoft definitions when it comes to incidental details like parameter naming (so that users looking at that documentation can know what to expect), or if we should take responsibility for our own parameter naming conventions.

I lean toward the latter, but being able to duplicate function declarations from spec documents directly is helpful for making sure we expose exactly the right signature.